### PR TITLE
Adding Datastore repo

### PR DESCRIPTION
**Pull request by pranjal947**

### CODE REVIEW

File: repository/cassandra.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+			}
+
+			shardID := int(clusterID) / c.shardDivisor
+			dataFields["shard_id"] = shardID
+		}
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			queryStmt := `INSERT INTO ` + c.keyspace + `.` + tableName + ` (` + strings.Join(onlyColumn, ",") + `)` + ` VALUES (` + strings.Join(qm, ",") + `)` + `;`
+			query := c.session.Query(queryStmt, args...)
```
Comment (hafidhfinandry) [inline repository/cassandra.go:81]: do we need to implement Cassandra library inside session? @pranjal947

**Review by pranjalmohansaxena — APPROVED** [review]

File: repository/cassandra.go (hunk 1 above)
Comment (pranjal947) [reply repository/cassandra.go:81]: Cassandra session will be initialised in init layer and then will be used in repository.

File: repository/cassandra.go [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+// 1. Key: ColumnField
+// 2. Value: ColumnValue
+func (c *cassandraRepo) Store(data []map[string]interface{}) (err error) {
+
+	var wg sync.WaitGroup
+	for _, dataRow := range data {
+		var columnName []string
+		var onlyColumn []string
+		var args []interface{}
+		var qm []string
```
Comment (shivanivarshney3) [inline repository/cassandra.go:55]: what does qm stands for, can we use clear naming

File: repository/cassandra.go [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+	}
+	wg.Wait()
+	return err
+}
+
+// Validating the Cassandra Parameters
+func validateAndProcessCassandraParams(params CassandraParams) (err error) {
+
+	var InvalidCassandraRL = errors.New("invalid_rate_limiter_for_cassandra")
+	var InvalidClusterKey = errors.New("invalid_cluster key")
```
Comment (shivanivarshney3) [inline repository/cassandra.go:96]: missed underscore on clusterkey

File: repository/cassandra.go [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+
+	CassandraParams struct {
+		ShardDivisor int
+		TableName    string
+		ClusterKey   string
+		Rl           ratelimit.Limiter
+	}
+)
+
+func NewCassandraRepo(param CassandraParams) (DatastoreRepository, error) {
```
Comment (shivanivarshney3) [inline repository/cassandra.go:27]: please add initialisation in main file

File: repository/cassandra.go (hunk 1 above)
Comment (hafidhfinandry) [reply repository/cassandra.go:81]: ok then, because I don't see session variable inside cassandraRepo and we don't import some kind of gocql library

Comment (shivanivarshney3) [reply repository/cassandra.go:81]: right, how can we use c.session in this then, we dont have it yet

**Pull request closed without merging**